src/selectors/per-thread/composed.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { createSelector } from 'reselect';
+
+import { tabSlugs, type TabSlug } from '../../app-logic/tabs-handling';
+
+import type { Selector } from '../../types/store';
+import type { $ReturnType } from '../../types/utils';
+import type { Thread, JsTracerTable } from '../../types/profile';
+
+/**
+ * Infer the return type from the getStackAndSampleSelectorsPerThread function. This
+ * is done that so that the local type definition with `Selector<T>` is the canonical
+ * definition for the type of the selector.
+ */
+export type ComposedSelectorsPerThread = $ReturnType<
+  typeof getComposedSelectorsPerThread
+>;
+
+type NeededThreadSelectors = {
+  getThread: Selector<Thread>,
+  getIsNetworkChartEmptyInFullRange: Selector<boolean>,
+  getJsTracerTable: Selector<JsTracerTable | null>,
+};
```

Comment:
I decided to use a partial type with only the needed function, because it was a bit too painful to get the full type.
Pro: easier for now.
Con: as we add more functions to this composed selectors we'll have to augment this type.

Reply:
I'm fine with this. Might be nice to explicitly add a short inline comment about why it's not a strict object, especially since this is usually just an oversight.

Reply:
good idea!
```
/**
 * This type contains the selectors needed for the extra selectors defined in
 * this file. It's non-exact because the passed object _will_ contain more
 * elements that we don't use here, and that's OK.
 */
```